packages/core-data/src/selectors.ts

```diff
@@ -55,27 +69,24 @@ interface UndoState extends Array< Object > {
 }
 
 interface UserState {
-	queries: Record< string, RecordKey[] >;
-	byId: Record< RecordKey, User< 'edit' > >;
+	queries: Record< string, GenericRecordKey[] >;
+	byId: Record< GenericRecordKey, User< 'edit' > >;
 }
 
-type RecordKey = number | string;
+type GenericRecordKey = number | string;
```

Comment:
I retained this for now, but ideally we'll get rid of this type in a follow-up PR that will clean up the types in this file and add any missing signatures.